platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/HalfDiskHashMap.java

```diff
@@ -245,6 +245,8 @@ public HalfDiskHashMap(
             numOfBuckets = Integer.highestOneBit(minimumBuckets) * 2;
             // we are new so no need for a loadedDataCallback
             loadedDataCallback = null;
+            // write metadata
+            writeMetadata(storeDir);
```

Comment:
This looks like a change compared to the previous behavior that is not explicitly listed in the PR description. Is this change intentional? If so, it would be nice to expand the comment at line 248 and indicate a reason why we want to (re-)write the metadata here.

It's a bit scary to see a write() method in a ctor that generally is responsible for loading/reading only. I'm not sure how well this would work in general, or, say, in unit and other tests if they happen to call this ctor.

Reply:
Yes, this change is intentional and related to the fix. I would say, it's been a bug in HDHM from the very beginning, just not discovered until the current PR. In the tests, an empty data source is created. Previously, it used long keys, and therefore there was a LongList index, but now with generic keys a new HDHM is created instead. When such a data source is loaded back from disk, LongList was ready to handle missing files, while HDHM complained about missing metadata file, which resulted in test failures. The fix is to force writing the metadata file, when a new HDHM is created.

Reply:
Got it. Sounds good. Thanks!